core/domain/prod_validation_jobs_one_off.py

```diff
@@ -5084,6 +5084,64 @@ def _get_custom_validation_functions(cls):
             cls._validate_score]
 
 
+class PendingDeletionRequestModelValidator(BaseUserModelValidator):
+    """Class for validating PendingDeletionRequestModels."""
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        return {
+            'user_settings_ids': (user_models.UserSettingsModel, [item.id]),
+        }
+
+    @classmethod
+    def _validate_explorations_are_marked_deleted(cls, item):
+        """Validates that explorations for model are marked as deleted.
+
+        Args:
+            item: ndb.Model. BaseUserModel to validate.
+        """
+        exp_ids = cls._get_exp_ids(item)
+
+        not_marked_exp_ids = [
+            exp_id for exp_id in exp_ids if
+            not exp_models.ExplorationModel.get_by_id(exp_id).deleted]
```

Comment:
The test will fail here if the model returned is None, right (None.deleted will give error). So, shouldn't this be:

```
exp = exp_models.ExplorationModel.get_by_id(exp_id)
if exp == None or not exp.deleted:
  ....
```

Reply:
Yes, makes sense, done.